ksqldb-functional-tests/src/test/resources/query-validation-tests/collect-set.json

```diff
@@ -8,6 +8,86 @@
     "a join or another aggregate."
   ],
   "tests": [
+    {
+      "name": "collect_set with arrays, structs, and maps",
```

Comment:
can we add tests for both of these for decimal type as well (and nested decimal). We should make sure that the set membership check works. 

Also what if we have a struct with a `null` value in one of the fields (we should make sure the set membership check works there as well)? What about a `null` at the top level?